File: DS-Algo-OU/Algorithms/MergeSort/MergeSort.c
#include <stdio.h>
#include <stdlib.h>

const int size = 4;

// declaring array for storing the data
int Array[] = {40, 30, 20, 10};

// declaring array to store the sorted array
int Sorted_array[size];

int *MergeSort(int array[], int left, int right)
{
     if (left > right)
     {
          return array;
     }

     // divide the array

     // 1) split the array into two parts

     // declaring mid to find out the mid point of the array
     int mid = (left + right) / 2;

     // declaring arrays for left and right
     static int left[mid];
     static int right[right - mid];

     int index_for_left_array = 0;
     int index_for_right_array = 0;

     int index_for_main_array = 0;
     while (index_for_left_array < mid && index_for_main_array < size)
     {
          left[index_for_left_array] = array[index_for_main_array];
     }

     
    while(index_for_right_array < (right-mid) && index_for_main_array < size)
    {
         right[index_for_right_array] = array[index_for_main_array];
    }
    
    // 2) calling left and right array for recursion
    MergeSort(left, left, mid);
    MergeSort(right, mid+1, right);

    // 3) sort the array
}

File: DS-Algo-OU/Algorithms/Sorting/MergeSort/MergeSort.py
# Write a python program to implement MergeSort

# 1) take an array
array = [5,4,3,2,1]

# Merge the divided arrays
def Merge(array, left, mid, right):

    

# 2) Write a function to divide the array
def MergeSort(array, left, right):

    if(left < right):
        # find out mid
        mid = (left + right) // 2

        # split the array into two parts
        MergeSort(array, left, mid)
        MergeSort(array, mid+1, right)
    

    